ecosystem_cicd_tools/new_cicd/actions.py

```diff
@@ -45,6 +46,51 @@ def upload_assets_to_release(assets, release_name, repository, **_):
             os.environ.get('CIRCLE_USERNAME', 'earthmant')
         )
 
+    # Time to wait for the plugin to load
+    # And checking that everything was updated correctly
+    max_time = 180  # it should not take longer than 180 seconds.
+    min_time = 20  # It should definitely take longer than 20 seconds.
+    interval = 10  # We check every 10 seconds.
+    current = 0
+    while True:
+        if current < min_time:
+            current += interval
+            continue
+        if max_time <= current:
+            raise RuntimeError(
+                'Timed out waiting for marketplace plugin update.')
+        if checking_the_upload_of_the_plugin(release_name,
+                                             release,
+                                             repository):
+            break
+        sleep(interval)
+        current += interval
+
+
+def checking_the_upload_of_the_plugin(release_name, release, repository, **_):
+
+    name_plugin = repository.name
+    plugin_id = marketplace.get_plugin_id(name_plugin)
+    version = marketplace.get_plugin_versions(plugin_id)[-1]
+    node_types = marketplace.get_node_types_for_plugin_version(name_plugin,
+                                                               version)
+    assets = release.get_assets()
+    assets_list = []
+
+    for asset in assets:
+        assets_list.append(asset.label)
+
+    if 'plugin.yaml' not in assets_list:
+        raise RuntimeError(
+            'Failed to update marketplace with plugin release.'
+            'The plugin.yaml file does not exist')
+
+    if release_name not in version and not node_types:
+        raise RuntimeError(
+            'Failed to update marketplace with plugin release.')
```

Comment:
Let's check this before `release.get_assets()` and also before `     node_types = marketplace.get_node_types_for_plugin_version(name_plugin,`